tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/NetStandardLibraries/CosmosDB/CosmosDBExerciser.cs

```diff
@@ -39,61 +40,95 @@ public class CosmosDBExerciser
+                    while (iterator.HasMoreResults)
                     {
-                        Console.WriteLine(db.Id);
+                        using ResponseMessage response = await iterator.ReadNextAsync();
+                        using (StreamReader sr = new StreamReader(response.Content))
+                        {
+                            var a = sr.ReadToEnd();
+                            Assert.False(string.IsNullOrEmpty(a));
+                        }
                     }
                 }
             }
+            finally
+            {
+                // Delete the database from Azure Cosmos.
+                await database.DeleteAsync();
+            }
+        }
+
+        [LibraryMethod]
+        [Transaction]
+        [MethodImpl(MethodImplOptions.NoOptimization | MethodImplOptions.NoInlining)]
+        public async Task CreateReadAndDeleteContainers(string databaseId, string containerId)
+        {
+            var endpoint = CosmosDBConfiguration.CosmosDBServer;
+            var authKey = CosmosDBConfiguration.AuthKey;
+
+            using var client = new CosmosClient(endpoint, authKey, _cosmosClientOptions);
+            var databaseResponse = await client.CreateDatabaseIfNotExistsAsync(databaseId);
+            var database = databaseResponse.Database;
 
-            // Read database using GetDatabaseStreamQueryIterator
-            using (FeedIterator iterator = client.GetDatabaseQueryStreamIterator())
+            try
             {
-                while (iterator.HasMoreResults)
+                await database.CreateContainerAsync(containerId, "/pk");
+
+                string queryText = "SELECT * FROM c";
+                using FeedIterator<ContainerProperties> feedIterator = database.GetContainerQueryIterator<ContainerProperties>(queryText);
+                while (feedIterator.HasMoreResults)
                 {
-                    using ResponseMessage response = await iterator.ReadNextAsync();
-                    using (StreamReader sr = new StreamReader(response.Content))
+                    var r = await feedIterator.ReadNextAsync();
+
+                    foreach (var c in r)
                     {
-                        sr.ReadToEnd();
+                        var container = database.GetContainer(c.Id);
+                        await container.ReadContainerAsync();
+                        await container.DeleteContainerAsync();
                     }
                 }
             }
-
-            // Delete the database from Azure Cosmos.
-            await database.DeleteAsync();
+            finally
+            {
+                await database.DeleteAsync();
+            }
         }
 
-
         [LibraryMethod]
         public static void StartAgent()
         {
             NewRelic.Api.Agent.NewRelic.StartAgent();
             //Get everything started up and time for initial Sample().
             Thread.Sleep(TimeSpan.FromSeconds(10));
         }
-
```

Comment:
We no longer need this as the `DynamicMethodInvoker` can now handle `Task` based method.